.github/workflows/build-rapids.yml

```diff
@@ -0,0 +1,111 @@
+name: Build all RAPIDS repositories
+
+on:
+  workflow_call:
+
+jobs:
+  check-event:
+    name: Check GH Event
+    runs-on: ubuntu-latest
+    outputs:
+      ok: ${{ steps.check_gh_event.outputs.ok }}
+    steps:
+      - id: check_gh_event
+        name: Check GH Event
+        shell: bash
+        run: |
+          [[ '${{ github.event_name }}' == 'push' && '${{ github.repository }}' == 'NVIDIA/cccl' ]] || \
+          [[ '${{ github.event_name }}' == 'schedule' && '${{ github.repository }}' == 'NVIDIA/cccl' ]] || \
+          [[ '${{ github.event_name }}' == 'pull_request' && '${{ github.repository }}' != 'NVIDIA/cccl' ]] \
+          && echo "ok=true"  | tee -a $GITHUB_OUTPUT \
+          || echo "ok=false" | tee -a $GITHUB_OUTPUT;
+
+  build-rapids:
+    name: Build RAPIDS (${{ matrix.libs }})
+    if: needs.check-event.outputs.ok == 'true'
+    needs: check-event
+    secrets: inherit
+    # TODO: this is temporary
+    uses: ./.github/workflows/build-in-devcontainer.yml
+    # uses: rapidsai/shared-workflows/.github/workflows/build-in-devcontainer.yaml@branch-24.06
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { libs: 'rmm KvikIO cudf cudf_kafka cuspatial',         args: 'BUILD_TESTS BUILD_BENCHMARKS' }
+          - { libs: 'rmm ucxx raft cuvs',                           args: 'BUILD_TESTS BUILD_BENCHMARKS BUILD_ANN_BENCH BUILD_PRIMS_BENCH' }
+          - { libs: 'rmm ucxx raft cumlprims_mg cuml',              args: 'BUILD_TESTS BUILD_BENCHMARKS BUILD_ANN_BENCH BUILD_PRIMS_BENCH' }
+          - { libs: 'rmm ucxx raft cugraph-ops wholegraph cugraph', args: 'BUILD_TESTS BUILD_BENCHMARKS BUILD_ANN_BENCH BUILD_PRIMS_BENCH BUILD_CUGRAPH_MG_TESTS' }
+    permissions:
+      actions: read
+      packages: read
+      id-token: write
+      contents: read
+      pull-requests: read
+    with:
+      dir: 'ci/rapids'
+      arch: '["amd64"]'
+      cuda: '["12.2"]'
+      pkgr: '["conda"]'
+      node_type: cpu32
+      extra-repo-deploy-key: RAPIDSAI_CUMLPRIMS_DEPLOY_KEY
+      extra-repo-deploy-key-2: RAPIDSAI_CUGRAPH_OPS_DEPLOY_KEY
+      env: |
+        CI=true
+        RAPIDS_LIBS=${{ matrix.libs }}
+        RAPIDS_TEST_OPTIONS=${{ matrix.args }}
+        RAPIDS_cmake_GIT_REPO={"upstream": "trxcllnt", "tag": "fea/cccl-2.5"}
+        RAPIDS_cudf_GIT_REPO={"upstream": "rapidsai", "tag": "branch-24.06"}
```

Comment:
Let's move all these to 24.08 (and ucxx 0.39).

Reply:
Done